ParadiseSS13/Paradise
# APC & Machinery Refactor + Local Powernets

## What Does This PR Do
First step in #19786

- Refactors APCs (Moving stuff to defines, splitting apc.dm into like 4 different files)
- Moves power processing in areas to a local powernet datum as opposed to on the `/area` or `/machinery/power/apc` level
- All machinery will now use a passive (and more CPU performative) method for power consumption where it used to consume power every process interval
- Refactors power_change() to only allow icon/lighting updating when the machine's NOPOWER flag changes
- Standardizes power proc/var usage across machines

Note that this does not rebalance power values nor does it actually have any player facing changes

## Why It's Good For The Game
Power code is mostly baycode. Furthermore it's only slightly less magical and confusing than atmos code so it needs to be brought up to modern standards before I even think about adding/modifying power features (I learned my lesson from econ rework)

## Testing
- Compiled and Ran
- Check, flicked, and played with all buttons and options on APCs
- Depowered the entire station and repowered it
- Fully constructed and deconstructed APCs
- Set up Supermatter Engine and ensured power was flowing correct into SMES and then being delivered to APCs
- Checked to ensure that APCs and light switches can be built properly in newly blueprinted areas
- ran on personal server for an hour to observe possible long-term power issues


no player facing changes so no changelog
